Transfer Studio how-to use case to SP guide (domain model) (#5988)
mendix/docs: content/en/docs/refguide/modeling/domain-model/create-a-basic-data-layer.md

Delete behavior includes objects which are in memory. This means that cascading delete or prevention of delete applies, even if the associated object has not been committed.
{{% /alert %}}

> ## 7 An Example of a Complete Use Case {#example-use-case}

Reviewer: I would suggest a different title, the section does not present a complete use case, it presents an example briefly. "Defining Data for a Domain Model" or something similar would fit for the section contents better.

Author: I changed the title to "An Example for Defining a Domain Model" for section 7, and added the following three sub-sections: "Defining What Data to Include", "Defining Entities", and "Defining Associations". The main changes are in this commit: https://github.com/mendix/docs/pull/5988/commits/3049e3309075600585efe6f9541b12663599a34e (sorry about the name of this commit, it should be "Address" not "Add". And the lower-case "example" in the title for section 7 has been fixed to "Example" in another commit.

> In the above sections, you learn the basics of how to configure a domain model. In the following example, a complete use case is presented: you are configuring the domain model for an online shopping app.

Reviewer: Users are not configuring anything in this section, this is a description of how to define data.

An example of the domain model for the online shopping app is shown below:

> {{< figure src="/attachments/refguide/modeling/domain-model/create-a-basic-data-layer/domain-model-online-shop.png" alt="Domain Model online shopping app" >}}

Reviewer: Would it be a good idea to introduce Section 3 from Studio here as a sub-section here? Showing the screenshot above at the end of it. 
The rest of the information can be a second sub-section called "Defining Entities" or something like this.

Author: I think the original Section 3 from Studio is a bit redundant regarding the details of each step, so I added this section but simplified the content. A brief introduction to the workflow is indeed still useful I think. What do you think?

Reviewer: @Yiyun333 , I think one of the points is to show the details to indicate where attributes are coming from, otherwise we have explained how to define only entities here.

Author: Good point! I've copied the original section back :)

For this online shopping app, you need to define the following information and create the corresponding entities for the domain model:

* The customer information is presented by the following entity:

>     * **Customer** - general information about the customer, such as their name, address, email, and so on

Reviewer: ```suggestion
    * **Customer** – general information about the customer, such as their name, address, email, and so on
```

* The product information is divided into the following entities:

    * **Product** – general information about the product, such as its name, description, price, and so on

>     * **Product_Image** – each product has an image, but you do not create it as an attribute. You need to create a special type of entity that allows you to store images. For more information, see [Image Uploader](/refguide/image-uploader/)

Reviewer: In Studio Pro this is not a special type of entity, it is a specialization that you need to create from System.

Author: Fixed: https://github.com/mendix/docs/pull/5988/commits/3049e3309075600585efe6f9541b12663599a34e

* The order information is divided into the following entities:

    * **Order** – general information about the order, such as its status, order number, name of the customer and their address, and so on
    * **Order line** – items ordered, their quantity and price
    * **Order confirmation** – confirmation that is sent to the customer that the order is placed

> You also need to define how these entities are connected to each other. For more information, see [Associations](/studio/domain-models-association-properties/). The following explains how each entity is associated:

Reviewer: There should be no links to Studio.

> * **Product_Image** and **Product** have a one-to-one association - one product image is connected to only one product

Reviewer: We use a dash, not a hyphen for lists.
```suggestion
* **Product_Image** and **Product** have a one-to-one association – one product image is connected to only one product
```

> * **Order** and **Customer** have a one-to-many association - an order is placed by a customer. Several orders can be connected to one customer

Reviewer: When using several full sentences in a list, we then need to capitalize the it as a sentence and use periods for every list item. For example,
* **Order** and **Customer** have a one-to-many association - An order is placed by a customer. Several orders can be connected to one customer.
* **Product** and **Order_Line** have a one-to-many association - The **Order_Line** uses information about the product. One product can be associated with several order lines.

Author: Thanks for the clarification! Also fixed in the doc in the same commit as above.

Delete behavior includes objects which are in memory. This means that cascading delete or prevention of delete applies, even if the associated object has not been committed.
{{% /alert %}}

> ## 7 An Example for Defining a Domain Model {#example-domain-model}

Reviewer: ```suggestion
## 7 An Example of Defining a Domain Model {#example-domain-model}
```

* Customer
* Product
* Order

After you have decided what information to include in your domain model, you still need to decide what kind of entities should be created in the domain model that can represent the information, and how these entities are connected to each other. In the following sections, we explain how to define the entities and their associataions for this online shopping app.

An example of the domain model for the online shopping app is shown below:

{{< figure src="/attachments/refguide/modeling/domain-model/create-a-basic-data-layer/domain-model-online-shop.png" alt="Domain Model online shopping app" >}}

### 7.2 Defining Entities

 The following shows what entities you should create in your domain model for the online shopping app:

* The customer information should be presented by the following entity:

    * **Customer** – general information about the customer, such as their name, address, and email address

* The product information should be divided into the following entities:

    * **Product** – general information about the product, such as its name, description, and price

>     * **Product_Image** – each product has an image, but you do not create it as an attribute. You need to create an entity that is a specialization of the **System.Image** entity to allow you to store images. For more information, see [Image Uploader](/refguide/image-uploader/)

Reviewer: Image Uploader is one of the examples on how to use it on a page, but I am not sure that this is the only example of why you would use System.Image specialization. Perhaps, saying "for an example on how to display images on a page, see ..." is a bit better here?

Author: Also a good point! I added this to the doc: "For an example on how the **System.Image** entity is used for displaying images on a page, see ...".

### 7.1 Defining What Data to Include

Understanding the typical process helps you define what data to include to your domain model. The workflow for new customers of the online shopping app looks the following way:

1. A customer registers in the online shopping app and enters the following details:
    1. Full name
    2. Address
    3. Email
    4. Date of birth
2. When registration is complete, a *unique ID* is assigned to the customer. 
3. The customer browses through *products* and the following product details are displayed:
    1. Product image 
    2. Name
    3. Description 
    4. Availability
    5. Price 
    6. Vendor
    7. Unique product ID 
4. The customer adds products to a shopping cart. 
5. In the shopping cart, every item is presented as a separate line showing *quantity* and *price* per line. The customer checks the order, pays it, and gets a *confirmation* with the *order details* and the *date* the order is purchased on. 

Based on the description above, you can divide your data into the following elements: 

* Customer
* Product
* Order

> In the following sections, we explain how to represent the data you want to include in your app with entities and their attributes in the domain model and how you should set up the associations between the entities. 

Reviewer: "We" is more of an Academy style, not often used in Docs. 
```suggestion
The following sections explain how to represent the data you want to include in your app with entities and their attributes in the domain model and how you should set up the associations between the entities. 
```

An example of the domain model for the online shopping app is shown below:

{{< figure src="/attachments/refguide/modeling/domain-model/create-a-basic-data-layer/domain-model-online-shop.png" alt="Domain Model online shopping app" >}}

### 7.2 Defining Entities

 The following shows what entities you should create in your domain model for the online shopping app:

* The customer information should be presented by the following entity:

    * **Customer** – general information about the customer, such as their name, address, and email address

* The product information should be divided into the following entities:

    * **Product** – general information about the product, such as its name, description, and price

>     * **Product_Image** – each product has an image, but you do not create it as an attribute. You need to create an entity that is a specialization of the **System.Image** entity to allow you to store images. For an example on how the **System.Image** entity is used for displaying images on a page, see [Image Uploader](/refguide/image-uploader/)

Reviewer: It is not System.Image that is used but its specialization, small correction there.

```suggestion
    * **Product_Image** – each product has an image, but you do not create it as an attribute. You need to create an entity that is a specialization of the **System.Image** entity to allow you to store images. For an example on how to display images on a page, see [Image Uploader](/refguide/image-uploader/)
```
